packages/typespec-autorest-canonical/README.md

```diff
@@ -0,0 +1,131 @@
+# @azure-tools/typespec-autorest-canonical
+
+TypeSpec library for emitting canonical version swagger
+
+## Install
+
+```bash
+npm install @azure-tools/typespec-autorest-canonical
+```
+
+## Emitter
+
+### Usage
+
+1. Via the command line
+
+```bash
+tsp compile . --emit=@azure-tools/typespec-autorest-canonical
+```
+
+2. Via the config
+
+```yaml
+emit:
+  - "@azure-tools/typespec-autorest-canonical"
+```
+
+### Emitter options
+
+#### `output-dir`
+
+**Type:** `string`
+
+Deprecated DO NOT USE. Use built-in emitter-output-dir instead
+
+#### `output-file`
+
+**Type:** `string`
+
+Name of the output file.
+Output file will interpolate the following values:
+
+- service-name: Name of the service if multiple
+- version: Version of the service if multiple
+- azure-resource-provider-folder: Value of the azure-resource-provider-folder option
+- version-status: Only enabled if azure-resource-provider-folder is set. `preview` if version contains preview, stable otherwise.
+
+Default: `{azure-resource-provider-folder}/{service-name}/{version-status}/canonical/openapi.json`
+
+Example: Single service no versioning
+
+- `canonical.openapi.json`
+
+Example: Multiple services no versioning
+
+- `Service1.canonical.openapi.json`
+- `Service2.canonical.openapi.json`
+
+Example: Single service with versioning
+
+- `canonical.openapi.json`
+
+Example: Multiple service with versioning
+
+- `Service1.canonical.openapi.json`
+- `Service2.canonical.openapi.json`
+
+#### `version`
+
+**Type:** `string`
+
+#### `azure-resource-provider-folder`
```

Comment:
And this one

Reply:
check every option, I am not sure you really need anything but maybe `new-line` specially if this is just going to be piped internally.

Reply:
updated options